### PR TITLE
feat: add custom header support

### PR DESCRIPTION
**Pull request by eshanholtz**
This PR adds support for passing custom Twilio-supported HTTP headers as request arguments.

### CODE REVIEW

File: lib/twilio-ruby/framework/version.rb [hunk 1]
```diff
@@ -127,7 +127,7 @@ def read_limits(limit = nil, page_size = nil)
         }
       end
 
-      def page(method, uri, params = {}, data = {}, headers = {}, auth = nil, timeout = nil)
+      def page(method, uri, params = {}, data = {}, headers: {}, auth: nil, timeout: nil)
```
Comment (childish-sambino) [inline lib/twilio-ruby/framework/version.rb:130]: Just for clarity, this is backwards-compatible in Ruby, correct?

File: spec/integration/studio/v1/flow/execution_spec.rb [hunk 2]
```diff
@@ -104,6 +106,7 @@
     }.to raise_exception(Twilio::REST::TwilioError)
 
     values = {'To' => '+15558675310', 'From' => '+15017122661', }
+    headers = {}
```
Comment (childish-sambino) [inline spec/integration/studio/v1/flow/execution_spec.rb:109]: Seeing a lot of these that are unused.

Comment (eshanholtz) [reply spec/integration/studio/v1/flow/execution_spec.rb:109]: Pushing update to add an additional check when rendering these

File: lib/twilio-ruby/framework/version.rb (hunk 1 above)
Comment (eshanholtz) [reply lib/twilio-ruby/framework/version.rb:130]: Ugh, no, it's not. Fixing.

**Pull request merged**